.github/workflows/tests.yml

```diff
@@ -0,0 +1,63 @@
+
+name: Tests
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - "tests/**"
+      - "src/**"
+      - "pyproject.toml"
+      - "poetry.lock"
+      - ".github/workflows/tests.yml"
+  pull_request:
+    branches: [main]
+    paths:
+      - "tests/**"
+      - "src/**"
+      - "pyproject.toml"
+      - "poetry.lock"
+      - ".github/workflows/tests.yml"
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+  merge_group:
+
+permissions:
+  contents: read
+
+jobs:
+  
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Minimum supported; Maybe add in some newer version to test things won't break if used on newer versions of python?
+        python_version: ["3.8"] #, "3.x"] 
+    concurrency:
+      group: tests-${{ github.ref }}-${{ matrix.python_version }}
+      cancel-in-progress: true
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Poetry --with dev
+        run: |
+          curl -sSL https://install.python-poetry.org | python -
+          echo "$HOME/.poetry/bin" >> $GITHUB_PATH
+      - name: Set up Python ${{ matrix.python_version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python_version }}
+          cache: poetry
+      - name: Install dependencies
+        run: |
+          poetry install
+      - name: Run tests
+        run: |
+          poetry run pytest tests
+      - name: Upload generated files
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: test-generated-files-${{ github.event_name }}-${{ matrix.python_version }}
+          path: tests/test_question_templates/question_generated_outputs/
+          if-no-files-found: ignore
+          retention-days: 5
```

Comment:
```suggestion

```
This one's on me for not mentioning - we don't need to upload generated files on test failures cause we never generate files in tests (for now at least)